spark/ingestion/src/main/scala/feast/ingestion/metrics/StreamingMetrics.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.ingestion.metrics
+
+import org.apache.spark.SparkEnv
+import org.apache.spark.metrics.source.StreamingMetricSource
+import org.apache.spark.sql.streaming.StreamingQueryProgress
+
+class StreamingMetrics extends Serializable {
+
+  private lazy val metricSource: Option[StreamingMetricSource] = {
+    val metricsSystem = SparkEnv.get.metricsSystem
+    StreamingMetricsLock.synchronized {
```

Comment:
no need to do lazy initialization, if metric are used on driver only

Reply:
You can simply register source through spark config (see https://spark.apache.org/docs/latest/monitoring.html#metrics)